### PR TITLE
Add design doc

### PR DESCRIPTION
**Pull request by ajeddeloh**
Please check for:
 - Grammar, clarity, etc
 - That I accurately recorded the discussion
 - Formatting seems good

I only did the partitioning issue. We can add summaries of other closed issues in more PRs.

I envision that when we want to close out an issue, we PR this doc with a summary. Once we agree the PR is good and accurately encapsulates the discussion, we merge the PR and close the issue.

As the doc grows we may want to break it out into multiple docs in a directory, but that can come later.

Finally, I'd prefer to keep this doc about design of FCOS itself and not about the process around it. We can have a seperate doc for that (the README does a great job right now) if we want.

### CODE REVIEW

**Comment by vtolstov** [comment]
sorry, bay be this is not proper place, do you think about steps that needs to be done to move from old atomic to fcos ?  Or process is simple - reformat/reinstall ? =)

**Comment by ajeddeloh** [comment]
Not really the proper place, but I'll answer anyway =)
See https://discussion.fedoraproject.org/t/launch-faq-how-do-i-migrate-from-fedora-atomic-host-to-fedora-coreos/51

tl;dr: You'll want to reprovision. For one, FCOS will use [Ignition](https://github.com/coreos/ignition) instead of cloudinit for one. Rebasing _might_ work but that's very much in the "proceed at your own risk" and completely unsupported.

It's still to early to tell what exactly will change, but reprovisioning will be your best bet regardless.

**Comment by vtolstov** [comment]
Thanks!

File: Design.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Design Document
+
+This document captures the design decisions made after dicussion in issues. When a design issue is closed, the
+conclusion should be summarized here with a link to the issue.
+
+## Disk Layout
+
+Originally discussed in issue #18
```
Comment (arithx) [inline Design.md]: Not sure if it's just a bug in the Rich Diff but it doesn't seem to be linking properly to #18
